### ISSUE TITLE
How to specify install path

### ISSUE THREAD

**Issue by Beep6581** [bootchk/resynthesizer#34]
I cloned to `~/programs/code-resynthesizer/` and want to install to `~/programs/resynthesizer/`
README says:
> Installs to the shared Gimp directories.  Except on Ubuntu, it installs to /usr/local/lib/gimp/2.0/plug-ins and you may need to set the GIMP Edit>Preferences>Folders>Plugins> to include that path.

How do I change that?
I tried `./configure --prefix=$HOME/programs/resynthesizer` but it still tried to install to the wrong folder and failed:
`/usr/bin/install: cannot create regular file '/usr/lib64/gimp/2.0/plug-ins/resynthesizer': Permission denied`

**Comment by bootchk** [comment]
The comment in the README is about a Ubuntu versus other distro issue:  I think most of the world installs plugins to /usr/lib/gimp.... but Ubuntu installs to /usr/local/lib/gimp....   IOW, an issue about politics and the "Filesystem Hierarchy Standard" (search for that term.)  Ubuntu wants to install anything that a user has downloaded or built to a segregated directory underneath /usr/local.  And GIMP doesn't know about that, so a GIMP user must change their preferences.

I think that is a separate issue from what --prefix does.  AFAIK --prefix just points to the directory where you are building, and is unrelated to where the build system installs the built product.

(Aside: the world is moving to a separate build directory from the source directory.  I don't think the resynthesizer project is set up that way.  The resynthesizer build system was copied from a GIMP plugin template, years ago.  With some subsequent maintenance of various build scripts.)

I think it is trying to install to the right place (for your non-Ubuntu distro), but it is a permissions issue.  Maybe 'sudo make install' would work?

I could be wrong, I am not an expert on build systems or permissions.

**Comment by Beep6581** [comment]
I don't want it installed to that (correct) place, I want the compiled file(s) moved to a folder of my choosing, so that I can easily examine them. I will also probably need to do this to make the Gentoo ebuild. How do I do that?

**Comment by bootchk** [comment]
I think...

in configure.ac, line 92:   gimpplugindir="$libdir/gimp/2.0"

This defines a variable that is used later in the scripts that install.
I don't know where the variable libdir is defined, but I guess the value is "/usr/bin".
I suggest you hack line 92, if it is just for your temporary use.

I wouldn't think that for the distribution, the installation place would be user configurable.  There are conventions for where things are installed.  But as I discussed earlier, conventions are not much use if e.g. Ubuntu diverges from some people's convention by installing to /usr/local/bin

Again, I didn't originate most of the build scripts, I copied them (and hacked them just a little) from the gimp plugin template: a dummy plugin, written in C, with build scripts that use
the gnu build system (tools automake, aclocal, M4, etc).
